src/Microsoft.ComponentDetection.Detectors/nuget/NuGetPackageReferenceFrameworkAwareDetector.cs

```diff
@@ -0,0 +1,231 @@
+namespace Microsoft.ComponentDetection.Detectors.NuGet;
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using global::NuGet.Packaging.Core;
+using global::NuGet.ProjectModel;
+using global::NuGet.Versioning;
+using Microsoft.ComponentDetection.Contracts;
+using Microsoft.ComponentDetection.Contracts.Internal;
+using Microsoft.ComponentDetection.Contracts.TypedComponent;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+public class NuGetPackageReferenceFrameworkAwareDetector : FileComponentDetector
+{
+    public const string OmittedFrameworkComponentsTelemetryKey = "OmittedFrameworkComponents";
+
+    public const string ProjectDependencyType = "project";
+
+    private readonly ConcurrentDictionary<string, int> frameworkComponentsThatWereOmmittedWithCount = new ConcurrentDictionary<string, int>();
```

Comment:
Are we using this dictionary at all in this new detector?

Reply:
Nope, I can remove it.  Let me see if there is more here to remove.